src/pgwire/src/message.rs

```diff
@@ -362,6 +362,9 @@ impl ErrorResponse {
             AdapterError::SqlCatalog(_) => SqlState::INTERNAL_ERROR,
             AdapterError::SubscribeOnlyTransaction => SqlState::INVALID_TRANSACTION_STATE,
             AdapterError::Transform(_) => SqlState::INTERNAL_ERROR,
+            AdapterError::UnallowedOnCluster { .. } => {
+                SqlState::S_R_E_PROHIBITED_SQL_STATEMENT_ATTEMPTED
+            }
```

Comment:
No worries! Looking through all of the error codes the following seem to be somewhat decent:

- `protocol_violation` - Class 08 — Connection Exception
- `invalid_parameter_value` - Class 22 — Data Exception (not sure what classifies as a "parameter" though)
- `system_error` - Class 58 — System Error (errors external to PostgreSQL itself)

`system_error` seems to be the most generic, so that's what I'm leaning towards, what do you think?

Reply:
Hmm, `invalid_param_value` wouldn't work because `param` refers to parameterized queries like `SELECT * FROM $1`.
I don't think `system_error` makes sense either because the error is not external to Materialize. `protocol_violation` refers to the pgwire protocol, which this query isn't in violation of.

I actually think the privilege error is the best one the more I think about it. There's nothing inherently wrong with the query, we have just made it forbidden when using a specific cluster.

Sorry to be nit-picky on this.